service/controller/clusterapi/v27/key/machine_deployment_provider.go

```diff
@@ -0,0 +1,32 @@
+package key
+
+import (
+	"encoding/json"
+
+	g8sv1alpha1 "github.com/giantswarm/apiextensions/pkg/apis/cluster/v1alpha1"
+	cmav1alpha1 "sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
+)
+
+func machineDeploymentProviderSpec(machineDeployment cmav1alpha1.MachineDeployment) g8sv1alpha1.AWSMachineDeploymentSpec {
+	return mustG8sMachineDeploymentSpecFromCMAMachineDeploymentSpec(machineDeployment.Spec.Template.Spec.ProviderSpec)
+}
+
+func mustG8sMachineDeploymentSpecFromCMAMachineDeploymentSpec(cmaSpec cmav1alpha1.ProviderSpec) g8sv1alpha1.AWSMachineDeploymentSpec {
```

Comment:
The naming here got a bit wild but it is not used from the outside. You yourself would only call `machineDeploymentProviderSpec(machineDeployment)`.